python/prima/__init__.py

```diff
@@ -150,7 +143,5 @@ def minimize(fun, x0, args=(), method=None, bounds=None, constraints=(), callbac
         nonlinear_constraint_function,
         callback,
         options,
-        nlconstr0,
-        m_nlcon,
-        f0,
+        nlconstr0
```

Comment:
Sorry, this is not what I meant. 

The evaluation of `f0` and `nlconstr0` must happen at the same place. Only COBYLA needs them. The others do not. 

Maybe we have to talk about this face to face.

Reply:
The evaluation of `nlconstr0` needs to happen in Python while processing nonlinear constraints so that the size of the provided lb/ub can be checked. In theory we move processing of the nonlinear constraints into C++ but it would make the code much more complicated.

Remember that the Python interface is expected to handle multiple nonlinear constraints, so it doesn't make sense to evaluate nlconstr0 and f0 "in the same place" since the former will be inside of a loop. At best we can evaluate f0 just outside of the loop, but this requires passing the function down a few levels and then passing the result back up, which seems unnecessary.

Sure we can talk about it face to face, I will email you.